### DISCUSSION TITLE
Fixes for storage mode going read-only not applied to 3.2/develop

### DISCUSSION THREAD

**Discussion by timw** [orientechnologies/orientdb#9800] [General]
The changes in 72e7d639a5f96be237dc49c43c815ef7c9255400 haven't been applied to develop/3.2, which causes issues like #9721 and #9498.

We've looked at pulling these fixes up to 3.2, but there are a bunch of changes in that commit (i.e. deleting auto sharding indexes) that we don't understand the connections between.

Would a PR that pulls the error handling up be accepted, or should it also include the thread interruption?
Should the other changes in 72e7d639a5f96be237dc49c43c815ef7c9255400 also be pulled up, or is there some other context behind them only being applied in 3.1?

**Comment by tglman** [comment]
Hi,

I will check and in case redo the fixes, some of the changes are actually already in 3.2.x just in a different form.

Regards
